service/src/main/java/com/shortlink/service/server/UrlConversionServer.java

```diff
@@ -1,19 +1,101 @@
 package com.shortlink.service.server;
 
+import com.shortlink.service.data.UrlMap;
+import com.shortlink.service.data.UrlMapRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.data.redis.core.RedisTemplate;
 import org.springframework.stereotype.Component;
 
 @Component
 public class UrlConversionServer {
+    @Autowired
+    private RedisTemplate<String, Object> redisTemplate;
+
+    @Autowired
+    private UrlMapRepository urlRepository;
```

Comment:
This should be used in the Service

Reply:
The cache and DB logic are moved into "Services"